components/auth-nav.tsx

```diff
@@ -1,41 +1,64 @@
 "use client"
 
-import React from "react"
+import React, { useCallback, useEffect, useState } from "react"
 import Link from "next/link"
 import { useRouter } from "next/navigation"
 import {
   createClientComponentClient,
   type Session,
 } from "@supabase/auth-helpers-nextjs"
 
-import { cn } from "@/lib/utils"
+import { ProfileState } from "@/types/users"
+import { cn, getCountryFlag } from "@/lib/utils"
 import { Button, buttonVariants } from "@/components/ui/button"
 
+import { Skeleton } from "./ui/skeleton"
+
 export default function AuthNav({ session }: { session: Session | null }) {
   const router = useRouter()
   const supabase = createClientComponentClient()
+  const [detail, setDetail] = useState<ProfileState>()
 
-  const handleSignOut = async () => {
+  const handleSignOut = useCallback(async () => {
     await supabase.auth.signOut()
+    setDetail(undefined)
     router.refresh()
-  }
+  }, [router, supabase])
+
+  useEffect(() => {
+    const loadProfile = async (id: string | undefined) => {
+      const userProfile = await fetch(`/api/user/profile?id=${id}`)
+      const { data: profile } = await userProfile.json()
+      setDetail(profile)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    if (detail === undefined && session !== null) {
+      loadProfile(session?.user.id)
+    }
+  })
```

Comment:
```suggestion
  }, [session, details])
```

Reply:
I try to use this approach and gives me an infinite loop. The final approach it's my best try to avoid it.